modules/service/src/main/scala/ch/epfl/bluebrain/nexus/iam/service/groups/UserGroupsOps.scala

```diff
@@ -0,0 +1,29 @@
+package ch.epfl.bluebrain.nexus.iam.service.groups
+
+import ch.epfl.bluebrain.nexus.commons.iam.auth.{AuthenticatedUser, User}
+import ch.epfl.bluebrain.nexus.commons.iam.identity.Identity.GroupRef
+
+object UserGroupsOps {
+
+  /**
+    * Syntax sugar to expose methods on ''User''
+    * @param user the user
+    */
+  implicit class UserGroupsSyntax(user: User) {
+
+    /**
+      * Add ''usedGroups'' to the ''user'' and remove the rest
+      *
+      * @param usedGroups the groups to add to the current user. The rest will be removed
+      */
+    def filterGroups(usedGroups: Set[GroupRef]): User =
```

Comment:
would you want to keep the intersection of the two sets?

Reply:
it's also a bit unsafe to return a modified user in this case, as you're changing the implied semantics brought by the type